app/services/katello/pulp3/package_group.rb

```diff
@@ -0,0 +1,67 @@
+module Katello
+  module Pulp3
+    class PackageGroup < PulpContentUnit
+      include LazyAccessor
+
+      lazy_accessor :optional_package_names, :mandatory_package_names,
+                    :conditional_package_names, :default_package_names, :_id,
+                    :repository_memberships,
+                    :initializer => :backend_data
+
+      def _id
+        backend_data['pulp_href']
+      end
+
+      # Package type is now an integer:
+      # 1. default
+      # 2. optional
+      # 3. conditional
+      # 4. mandatory
+      # 5. unknown
+      # https://github.com/rpm-software-management/libcomps/blob/01a4759894cccff64d2561614a58281adf5ce859/libcomps/src/comps_docpackage.h#L36
+      def default_package_names
+        package_names_of_type(1)
+      end
+
+      def optional_package_names
+        package_names_of_type(2)
+      end
+
+      def conditional_package_names
+        package_names_of_type(3)
+      end
+
+      def mandatory_package_names
+        package_names_of_type(4)
+      end
+
+      def unknown_package_names
+        package_names_of_type(5)
+      end
+
+      def package_names_of_type(type)
+        packages = []
+        backend_data['packages'].each { |package| packages << package['name'] if package['type'] == type }
```

Comment:
How about something like this to avoid initializing the empty array?
```ruby
def package_names_of_type(type)
  filtered_packages = backend_data['packages'].select { |package| package['type'] == type }
  filtered_packages.map { |package| package['name'] }
end
```

Reply:
Sure that works